server.js

```diff
@@ -0,0 +1,149 @@
+// START server config
+const express = require("express")
+const app = express()
+const bodyParser = require("body-parser")
+require('dotenv').config()
+// END server config
+
+// START Firestore init
+const admin = require('firebase-admin')
+const serviceAccount = {
+    "project_id": process.env.firestore_project_id,
+    "private_key": process.env.firestore_private_key.replace(/\\n/g, '\n'),
+    "client_email": process.env.firestore_client_email,
+}
+
+admin.initializeApp({
+    credential: admin.credential.cert(serviceAccount)
+})
+const db = admin.firestore();
+const increment = admin.firestore.FieldValue.increment(1);
+const decrement = admin.firestore.FieldValue.increment(-1);
+// END Firestore init
+
+// DEBUG init database data
+const course_info = require("./data/scheduler_prereq.json")
+
+app.use(bodyParser.urlencoded({ extended: false }))
+app.use(bodyParser.json())
+app.use(express.static(__dirname + "/public"))
+
+// START server routing
+const port = process.env.PORT || 3000;
+
+app.listen(port, function () {
+    console.log(`Server started on ${port}`)
+})
+
+app.get("/", (req, res) => {
+    res.sendFile(__dirname + "public/index.html")
+})
+
+app.get("/courses", async (req, res) => {
```

Comment:
For all of these functions, I would separate the request handling logic from the route definitions. See [this](https://github.com/acmutd/portal-backend/tree/dev/functions/src) GitHub repo for an example (`index.ts`).